src/components/ContactDetailsModal/ContactDetailsModal.js

```diff
@@ -0,0 +1,263 @@
+};
+
+const InputWrapper = styled.View`
+  flex-direction: row;
+  align-items: center;
+  margin-top: ${spacing.small}px;
+`;
+
+const FieldIcon = styled(CachedImage)`
+  width: 48px;
+  height: 48px;
+  margin-right: ${spacing.small}px;
+`;
+
+const ErrorMessage = styled(BaseText)`
+  margin-top: ${spacing.large}px;
+  text-align: center;
+  color: ${themedColors.negative};
+  ${fontStyles.small};
+`;
+
+export const LoadingSpinner = styled(Spinner)`
+  margin-top: ${spacing.large}px;
+  align-items: center;
+  justify-content: center;
+`;
+
+export const TitleWrapper = styled.View`
+  flex-direction: row;
+  justify-content: center;
+  align-items: center;
+  width: 100%;
+`;
+
+export const QRCodeButton = styled.TouchableOpacity`
+  position: absolute;
+  top: 0;
+  right: 0;
+`;
+
+
+export const QRCodeIcon = styled(Icon)`
+  color: ${({ color }) => color};
+  font-size: 20px;
+`;
+
+const renderContactInput = (
+  value: string,
+  onChangeText: (value: string) => void,
+  placeholder: string,
+  icon: string,
+  theme: Theme,
+) => (
+  <InputWrapper>
+    <FieldIcon source={icon} />
+    <TextInput
+      theme={theme}
+      inputWrapperStyle={{ flex: 1, paddingBottom: 0 }}
+      inputProps={{
+        value,
+        onChangeText,
+        placeholder,
+      }}
+    />
+  </InputWrapper>
+);
+
+const ContactDetailsModal = ({
+  isVisible,
+  onModalHide,
+  theme,
+  contact,
+  onSavePress,
+  isDefaultNameEns,
+  title,
+  contacts,
+  showQRScanner,
+}: Props) => {
+  const [addressValue, setAddressValue] = useState('');
+  const [nameValue, setNameValue] = useState('');
+  const [dirtyInputs, setDirtyInputs] = useState(false);
+  const [resolvingEns, setResolvingEns] = useState(false);
+  const [isScannerVisible, setIsScannerVisible] = useState(false);
+  const { walletIcon, personIcon } = images(theme);
+
+  // reset input value on default change
+  useEffect(() => {
+    setDirtyInputs(false);
+    setAddressValue(contact?.ethAddress || '');
+    const defaultName = isDefaultNameEns ? contact?.ensName : contact?.name;
+    setNameValue(defaultName || '');
+  }, [contact]);
+
+  useEffect(() => {
+    if (!dirtyInputs && (!isEmpty(nameValue) || !isEmpty(addressValue))) {
+      setDirtyInputs(true);
+    }
+  }, [nameValue, addressValue]);
+
+  const resolveEnsName = async () => {
+    if (resolvingEns
+      || !isEmpty(nameValue)
+      || !isValidAddress(addressValue)
+      || isEnsName(addressValue)) return;
+
+    setResolvingEns(true);
+    const ensName = await lookupAddress(addressValue);
+    if (ensName) setNameValue(ensName);
+    setResolvingEns(false);
+  };
+
+  useEffect(() => { resolveEnsName(); }, [addressValue]);
+
+  const resolveEnsAddress = async () => {
+    if (resolvingEns
+      || !isEmpty(addressValue)
+      || !isEnsName(nameValue)) return;
+
+    setResolvingEns(true);
+    const { receiver } = await getReceiverWithEnsName(nameValue, false);
+    if (receiver) setAddressValue(receiver);
```

Comment:
Nothing, let's say you entered ENS name that doesn't exist then it won't resolve. However, we can show an error, wdyt?

Reply:
I think even just an alert / toast or something would be nice - just because if i entered an incorrect ENS name or one that doesn't exist - but i think it's correct, i would probably just wait and then assume that something is wrong with the system...